fastly/errors.go

```diff
@@ -327,11 +328,23 @@ func NewHTTPError(resp *http.Response) *HTTPError {
 		return &e
 	}
 
+	// Save a copy of the body as it's read/decoded.
+	// If decoding fails, it can then be used (via addDecodeErr)
+	// to create a generic error containing the body's read contents.
+	var bodyCp bytes.Buffer
```

Comment:
I'm not sure we need to save a copy of the entire body in case of error.  Any decoding error "should" have enough context on its own.

Reply:
This assumes that the response body contains useful text (that's either not JSON or is malformed JSON) for end users. Such as in the contrived [test case](https://github.com/fastly/go-fastly/pull/386/files#diff-3e88c7dd341d9c3ca93ea88df31f1d05b321d559d0b21eaec23c6c7047fa4e2eR130-R134):

```
404 - Not Found:
    Title:  Undefined error
    Detail: THIS IS NOT JSON
```

I'm not sure of the utility of returning the decode error. Taking the same test example, the error message would become:

```
404 - Not Found:

    Title:  Undefined error
    Detail: invalid character 'T' looking for beginning of value
```